pkg/handler/repositories.go

```diff
@@ -465,6 +466,59 @@ func (rh *RepositoryHandler) bulkDeleteRepositories(c echo.Context) error {
 	return c.NoContent(http.StatusNoContent)
 }
 
+// SnapshotRepository godoc
+// @summary 		snapshot a repository
+// @ID				createSnapshot
+// @Description     Snapshot a repository if not already snapshotting
+// @Tags			repositories
+// @Param  			uuid            path    string                          true   "Repository ID."
+// @Success			204 "Snapshot was successfully queued"
+// @Failure      	400 {object} ce.ErrorResponse
+// @Failure      	404 {object} ce.ErrorResponse
+// @Failure      	500 {object} ce.ErrorResponse
+// @Router			/repositories/{uuid}/snapshot/ [post]
+func (rh *RepositoryHandler) createSnapshot(c echo.Context) error {
+	if err := CheckSnapshotAccessible(c.Request().Context()); err != nil {
+		return err
+	}
+	uuid := c.Param("uuid")
+	_, orgID := getAccountIdOrgId(c)
+	response, err := rh.DaoRegistry.RepositoryConfig.Fetch(orgID, uuid)
+	if err != nil {
+		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error fetching repository", err.Error())
+	}
+
+	repo, err := rh.DaoRegistry.Repository.FetchForUrl(response.URL)
+	if err != nil {
+		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error fetching repository uuid", err.Error())
+	}
+
+	inProgress, err := rh.DaoRegistry.TaskInfo.IsSnapshotInProgress(orgID, repo.UUID)
+	if err != nil {
+		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error checking snapshot task", err.Error())
+	}
+
+	if inProgress {
+		return ce.NewErrorResponse(http.StatusConflict, "Error snapshotting repository", "This repository is currently being snapshotted.")
+	}
+
+	var repoUpdate dao.RepositoryUpdate
+	status := config.StatusPending
+
+	repoUpdate = dao.RepositoryUpdate{
+		UUID:   repo.UUID,
+		Status: &status,
+	}
+
+	if err := rh.DaoRegistry.Repository.Update(repoUpdate); err != nil {
+		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error setting status for snapshotted repository", err.Error())
+	}
```

Comment:
currently the status is really just tied to introspection.  We need to figure this out longer term though.  I'll open a card for this, but i'd leave out the Update of status

Reply:
Derp, yeah I know this, that's what I get for pattern following!